server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPServer.java

```diff
@@ -31,42 +33,45 @@
 import org.apache.james.http.jetty.JettyHttpServer;
 import org.apache.james.lifecycle.api.Configurable;
 
-import com.google.common.base.Throwables;
+import com.github.fge.lambdas.Throwing;
 
 public class JMAPServer implements Configurable {
 
-    private final JettyHttpServer server;
+    private final Optional<JettyHttpServer> server;
 
     @Inject
     private JMAPServer(JMAPConfiguration jmapConfiguration,
                        AuthenticationServlet authenticationServlet, JMAPServlet jmapServlet, DownloadServlet downloadServlet, UploadServlet uploadServlet,
                        AuthenticationFilter authenticationFilter, UserProvisioningFilter userProvisioningFilter, DefaultMailboxesProvisioningFilter defaultMailboxesProvisioningFilter) {
-
-        server = JettyHttpServer.create(
+        if (jmapConfiguration.isEnabled()) {
+            server = Optional.of(JettyHttpServer.create(
```

Comment:
jmapConfiguration.isEnabled() is a boolean.

(It's Java8)

Optional semantic is not well suited for this. This would be IMO mindfuck.